conclave-host/src/main/kotlin/com/r3/conclave/host/internal/gramine/GramineEnclaveHandle.kt

```diff
@@ -137,6 +158,57 @@ class GramineEnclaveHandle(
         }
     }
 
+    private fun prepareCommandToRun(user: String, group: String, port: Int): List<String> {
+        val dockerCommand = getDockerCommand(user, group)
+        val gramineCommand = getGramineExecutable(enclaveMode)
+        val javaCommand = getJavaCommand(port)
+
+        val command = dockerCommand + gramineCommand + javaCommand
+        logger.debug("Docker command: ${command.joinToString(" ")}")
+        return command
+    }
+
+    private fun getJavaCommand(port: Int): List<String> {
+        return listOf(
+            "java",
+            "-cp",
+            GRAMINE_ENCLAVE_JAR,
+            "com.r3.conclave.enclave.internal.GramineEntryPoint",
+            port.toString()
+        )
+    }
+
+    private fun getDockerCommand(user: String, group: String): List<String> {
+        return listOf(
+            "docker",
+            "run",
+            "-u",
+            "$user:$group",
+            "--network",
+            "host",
+            "--device=/dev/sgx_enclave",
+            "--device=/dev/sgx_provision",
+            "-v",
+            "/var/run/aesmd:/var/run/aesmd",
+            "-i",
+            "--rm",
+            "-v",
+            "${workingDirectory.toFile().absolutePath}:$DOCKER_WORKING_DIR",
+            "-w", workingDirectory.toFile().absolutePath.mapWorkingDirectory(),
+            "--security-opt",
+            "seccomp=${workingDirectory.toFile().absolutePath}/$GRAMINE_SECCOMP",
+            dockerImageTag
+        )
+    }
+
+    private fun retrieveImageTagFromManifest(): String {
+        val manifest = Paths.get(
+            if (enclaveMode == EnclaveMode.SIMULATION) "$workingDirectory/$GRAMINE_MANIFEST" else "$workingDirectory/$GRAMINE_SGX_MANIFEST"
+        )
+        val parseResult = Toml.parse(manifest)
+        return parseResult.getString("host.env.CONCLAVE_DOCKER_IMAGE_TAG")!!
```

Comment:
The conclave-build tag should be pass into the jar manifest via gradle. @filipesoliveira has some ideas. I suggest the tag be written into the conclave-common.jar since it's needed by both the plugin and the host.

Reply:
Ok, so you mean the `conclave-common.jar` manifest then. The plugin would write in the `conclave-common.jar` manifest and then read it on the host side, right? Isn't a bit weird anyway to leave this information in a generic file, i.e. I see it nice to use the Gramine manifest as a mechanism also to pass all Gramine related host and enclave info. 
Not a real issue though.
Regarding the tag, as I asked above: is the generation of this hash tag part of this PR then? I'd use the list of packages + fat jar + manifest files to create it, if so.

Reply:
>The plugin would write in the conclave-common.jar manifest and then read it on the host side, right? 
No, our building of the SDK, i.e. conclave-common/build.gradle, will need to get the tag and insert it into the jar manifest. Since both the host and the plugin use conclave-common, they should then have access to this tag value.

>Isn't a bit weird anyway to leave this information in a generic file, i.e. I see it nice to use the Gramine manifest as a mechanism also to pass all Gramine related host and enclave info.
I disagree. This build tag is a Conclave implementation detail and not strictly a Gramine concern. Also, it requires parsing the TOML file and the extra dependency. Using the manifest avoids that.